### ISSUE TITLE
Android: Augustus crashs on startup

### ISSUE THREAD

**Issue by s-sirchia** [Keriew/augustus#1016]
Augustus crashs on Android 14 (Google Pixel Pro).

Tried Augustus 3.2, 4.0 and the latest beta.

They all crash after having selected the correct path for the game files, after some seconds of black screens.

I have attached the Augustus log.

Just for comparison the latest version of Julius for Android on Google Play Store works correctly on the same phone (tried after Augustus, and all tests were made by clearing the game and app files).
[augustus-log.txt](https://github.com/Keriew/augustus/files/13918682/augustus-log.txt)

**Comment by crudelios** [comment]
This is most likely a driver issue on google nexus.

I have a Samsung Galaxy S22 with Android 14 and get no such crashes. I used to get crashes when rotating the phone, but those stopped after an update.

The reason Augustus crashes and Julius doesn't is because Augustus uses the GPU for displaying the game, while Julius uses the CPU. You can check if Augustus v3.1.0, which still uses the CPU to draw the game, still works.

Apart from that, there isn't much we can do on our end to fix your issue. 😟

**Comment by s-sirchia** [comment]
I confirm that Augustus 3.1.0 works, as for the screenshot.
![Screenshot_20240112-213740](https://github.com/Keriew/augustus/assets/16668957/6335de2a-f30f-447e-8ee2-664a0e8b906d)

It's a pity that, as usual, Android versions and manufacturers end up screwing us final users of their products.

I will continue enjoying Augustus on PC and using the 3.1.0 on Android for my on-the-go Caesar cravings!

Hoping that it will be somehow fixed in the future, either by Google itself or by someone that will uncover how to solve it !

**Comment by s-sirchia** [comment]
Just for further reference I want to better define that the phone is Google Pixel 8 Pro, with its stock rom Android 14 to the latest January update.

**Comment by crudelios** [comment]
Did it work before?

**Comment by s-sirchia** [comment]
No. 3.2 and 4.0 have always had this problem since I first tried them on this phone

**Comment by crudelios** [comment]
I've since updated the latest unstable to the newest version of SDL.

Is there any improvement there?

On another note, do any of you guys know how to use android studio to debug the crash? Since it only happens in Pixel 8 and I don't have one, I can't really test anything.

Thanks!

**Comment by crudelios** [comment]
I've updated the android build stuff. While I doubt this will fix the issue, can you check please?

**Comment by juanfont** [comment]
@crudelios unfortunately I still see it crashing (which does not happens with Julius).

I checked from logcat and I see a major crash of the system. Can I share it with you privately?

**Comment by juanfont** [comment]
I've sent you a DM in Discord

**Comment by crudelios** [comment]
If anyone wants to help with this one, I need someone with pixel 8 pro and who cab use android studio to debug this issue.

**Comment by s-sirchia** [comment]
I might try to help with the tests, just difficult to do it live, but if i can do it by my own schedule and give you the logcat output etc, I might do it.

**Comment by crudelios** [comment]
Any help is welcome!

More than the log cat, what I'd really need is someone to run the game with their phone from android studio in debug mode. That would pinpoint the exact line in the code where the game crashes.

I'd do it myself if I had a pixel 8 Pro, but those aren't exactly cheap unfortunately!

**Comment by s-sirchia** [comment]
If you point me the version of the project to test, I can do that as well!

**Comment by crudelios** [comment]
If you can use android studio that'd be really helpful.

Just know that the thing is finicky and doesn't always like compiling the game at first.

If you want, here are the instructions for building for augustus (it's the same as for Julius):

https://github.com/bvschaik/julius/wiki/Building-for-Android

**Comment by s-sirchia** [comment]
@crudelios sorry for the long delay. I tried more than once to build the project but it always fails when trying to build the SDL libraries, so i didn't manage to run the app from the source code.

**Comment by 7ider5** [comment]
Hello,

Same behaviour here.

Phone : Pixel 8a
OS : LineageOS 21-20240906-microG-akita / Android 14
Augustus-log content 👍 INFO:  Augustus version 4.0.0, ARM64 build
INFO:  Running on: Linux 5.15.137-android14-11-gb572b1fac135-ab11919372
INFO:  Creating screen 1080 x 2400, fullscreen, driver: Android
INFO:  pixel format wanted SDL_PIXELFORMAT_RGBX8888 (2), got SDL_PIXELFORMAT_RGBX8888 (2)
INFO:  Creating renderer
INFO:  Loaded renderer: opengles2
INFO:  Maximum scale of 64 applied (requested: 100)
INFO:  Maximum scale of 64 applied (requested: 100)
INFO:  Render texture created (642 x 1428)

Augustus worked like a charm on my previous Pixel 4a :/

**Comment by jjbarr** [comment]
It is possible to get Augustus to work by opening developer options, going to ANGLE preferences, and setting Augustus to use ANGLE rather than the native GL driver. Apparently ANGLE doesn't have whatever issue is causing the problem.

**Comment by Sephirex95** [comment]
Regrettably, moving this to wontfix, at least for now since it seems to exceed our current capacity for fixes.
Unless the Keeper of Ancient Scrolls decides otherwise, we'll have to rely on Android updates to help with this issue.
